### PR TITLE
Update texcmp

### PR DESCRIPTION
**Pull request by edemaine**
I found that the existing docker rules for `texcmp` did not make a working installation of LaTeX; in particular, it was missing a lot of the recently added packages.  I updated to Ubuntu 17.10 and tweaked the package listings until things compiled again.  Now I can run `texcmp` again.

### CODE REVIEW

File: dockers/texcmp/Dockerfile [hunk 1]
```diff
@@ -11,13 +11,15 @@ RUN sed -i 's/^\(deb.*updates\)/#\1/' /etc/apt/sources.list \
...
     apt-get install -qy --no-install-recommends \
         ca-certificates \
-        etoolbox \
         ghostscript \
         imagemagick \
         nodejs \
         npm \
+        texlive-generic-recommended \
         texlive-fonts-recommended \
+        texlive-font-utils \
```
Comment (ylemkimon) [inline dockers/texcmp/Dockerfile]: Where does it need `texlive-font-utils`?

**Comment by ylemkimon** [comment]
Yes, I have checked, though `texlive-latex-extra` contains many other packages not used. But I think it's fine.

**Comment by edemaine** [comment]
Yeah, as far as I could tell, `etoolbox` is no longer a stand-alone package.
https://packages.ubuntu.com/search?searchon=contents&keywords=etoolbox.sty&mode=exactfilename&suite=artful&arch=any

**Comment by kevinbarabash** [comment]
Cool.  Thanks for updating this.

**Pull request merged**